### PR TITLE
LibJS: Convert remaining Object AOs as well as TypedArray and Atomics AOs to ThrowCompletionOr

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/TypedArray.cpp [hunk 1]
```diff
@@ -405,5 +416,4 @@ void TypedArrayBase::visit_edges(Visitor& visitor)
     JS_DEFINE_TYPED_ARRAY(ClassName, snake_name, PrototypeName, ConstructorName, ArrayType);
 JS_ENUMERATE_TYPED_ARRAYS
 #undef __JS_ENUMERATE
-
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/TypedArray.cpp:408]: We usually keep the whitespace between the closing namespace curly and the inner code, no?

File: Userland/Libraries/LibJS/Runtime/TypedArray.cpp [hunk 2]
```diff
@@ -140,56 +140,99 @@ static ThrowCompletionOr<void> initialize_typed_array_from_array_buffer(GlobalOb
...
+    // 12. If IsSharedArrayBuffer(srcData) is false, then
+    //     a. Let bufferConstructor be ? SpeciesConstructor(srcData, %ArrayBuffer%).
+    // 13. Else,
+    //    a. Let bufferConstructor be %ArrayBuffer%.
+    // 14. If elementType is the same as srcType, then
+    //    a. Let data be ? CloneArrayBuffer(srcData, srcByteOffset, byteLength, bufferConstructor).
+    // 15. Else,
+
+    // a. Let data be ? AllocateArrayBuffer(bufferConstructor, byteLength).
     auto data = ArrayBuffer::create(global_object, byte_length.value());
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/TypedArray.cpp:189]: Missing exception check

**Review by IdanHo — COMMENTED** [review]
LGTM, apart from a couple nitpicks

File: Userland/Libraries/LibJS/Runtime/TypedArray.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/TypedArray.cpp]: Definitely, clang-format does this when you format while having one too many curly braces somewhere, and doesn't add it back later. Fixed.

File: Userland/Libraries/LibJS/Runtime/TypedArray.cpp (hunk 2 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/TypedArray.cpp:189]: fixed!

**Pull request merged**